lab-edward-smith/lib/server.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const http = require('http');
+const url = require('url');
+const cowsay = require('cowsay');
+
+let server = http.createServer((req, res) => {
+  let urlObj = url.parse(req.url, true);
+  if (req.url === '/' && req.method === 'GET') {
+    res.writeHead(200, {'Content-Type':'text/plain'});
+    res.write('API Endpoints:\n/api/cowsay');
+    res.end();
+  } else if (urlObj.pathname === '/api/cowsay' && req.method === 'GET') {
+    if (urlObj.query.text) {
+      if (urlObj.query.cow) {
+        res.writeHead(200, {'Content-Type': 'text/plain'});
+        res.write(cowsay.say({
+          text: urlObj.query.text,
+          f: urlObj.query.cow
+        }));
+        res.end();
+        return;
+      }
+
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      res.write(cowsay.say({
+        text: urlObj.query.text,
+        f: 'default'
+      }));
+      res.end();
+    } else {
+      res.writeHead(400, {'Content-Type':'text/plain'});
+      res.write(cowsay.say({
+        text: 'bad request\ntry localhost:3000/api/cowsay?text=howdy'
+      }));
+      res.end();
+    }
+  } else if (urlObj.pathname === '/api/cowsay' && req.method === 'POST') {
+    if (urlObj.query.text) {
```

Comment:
When I try to make a POST request to api/cowsay with httpie, the request hangs. For a POST request, you don't want to grab the query property since the user will be sending the JSON over in the request body.